pkgs/by-name/cl/clash-verge-rev/package.nix

```
@@ -1,66 +1,173 @@
+    hash = "sha256-HyRTOqPj4SnV9gktqRegxOYz9c8mQHOX+IrdZlHhYpo=";
+  };
+  service = rustPlatform.buildRustPackage {
+    env = {
+      OPENSSL_NO_VENDOR = 1;
+    };
+    nativeBuildInputs = [
+      pkg-config
+      rustPlatform.cargoSetupHook
+      openssl
+    ];
+    buildInputs = [
+      openssl
+      webkitgtk
+    ];
+    inherit version;
+    src = src-service;
+    pname = "${pname}-service";
+    sourceRoot = "${src-service.name}";
+    cargoLock = {
+      lockFile = ./Cargo-service.lock;
+    };
+  };
+  webui = stdenv.mkDerivation {
+    inherit version src;
+    pname = "${pname}-webui";
+    pnpmDeps = pnpm.fetchDeps {
+      inherit pname version src;
+      hash = "sha256-DYsx1X1yXYEPFuMlvZtbJdefcCR8/wSUidFwsMy8oLk=";
+    };
 
-  nativeBuildInputs = [
-    dpkg
-    wrapGAppsHook3
-    autoPatchelfHook
-  ];
-
-  buildInputs = [
-    openssl
-    webkitgtk
-  ];
-
-  runtimeDependencies = [
-    (lib.getLib udev)
-    libayatana-appindicator
-  ];
+    nativeBuildInputs = [
+      nodejs
+      pnpm.configHook
+    ];
 
-  installPhase = ''
-    runHook preInstall
+    buildPhase = ''
+      runHook preBuild
 
-    mkdir -p $out/bin
-    mv usr/* $out
+      node --max_old_space_size=1024000 ./node_modules/vite/bin/vite.js build
 
-    runHook postInstall
-  '';
+      runHook postBuild
+    '';
 
-  postFixup = ''
-    rm -f $out/bin/verge-mihomo
-    ln -sf ${lib.getExe clash-meta} $out/bin/verge-mihomo
-  '';
+    installPhase = ''
+      runHook preInstall
 
-  passthru.updateScript = nix-update-script { };
+      cp -r dist $out
 
+      runHook postInstall
+    '';
+  };
   meta = {
     description = "Clash GUI based on tauri";
     homepage = "https://github.com/clash-verge-rev/clash-verge-rev";
     license = lib.licenses.gpl3Plus;
     mainProgram = "clash-verge";
-    maintainers = with lib.maintainers; [ Guanran928 ];
+    maintainers = with lib.maintainers; [
+      Guanran928
+      bot-wxt1221
+    ];
     platforms = [ "x86_64-linux" ];
-    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
   };
-})
+in
+rustPlatform.buildRustPackage {
+  inherit
+    pname
+    version
+    src
+    meta
+    ;
+
+  sourceRoot = "${src.name}/src-tauri";
+
+  cargoLock = {
+    lockFile = ./Cargo-tauri.lock;
+    outputHashes = {
+      "sysproxy-0.3.0" = "sha256-TEC51s/viqXUoEH9rJev8LdC2uHqefInNcarxeogePk=";
+    };
+  };
+  env = {
+    OPENSSL_NO_VENDOR = 1;
+  };
+  postPatch = ''
+    substituteInPlace ./tauri.conf.json \
+      --replace-fail '"distDir": "../dist",' '"distDir": "${webui}",' \
+      --replace-fail '"beforeBuildCommand": "pnpm run web:build"' '"beforeBuildCommand": ""'
+
+    mkdir -p ./sidecar
+    ln -s ${mihomo}/bin/mihomo sidecar/verge-mihomo-${rust.envVars.rustHostPlatform}
+    ln -s ${mihomo}/bin/mihomo sidecar/verge-mihomo-alpha-${rust.envVars.rustHostPlatform} # people should override mihomo to use alpha build. Because nixpkgs can't maintain alpha build.
```

```suggestion
    # people should override mihomo to use alpha build. Because nixpkgs can't maintain alpha build.
    ln -s ${mihomo}/bin/mihomo sidecar/verge-mihomo-alpha-${rust.envVars.rustHostPlatform}
```

Does the package even work without it?

---

This doesn't make sense. If there is no alpha version then just keep it missing.